validphys2/src/validphys/tests/photon/test_compute.py

```diff
@@ -58,29 +62,139 @@ def test_set_thresholds_alpha_em():
+                1
+            ],
+        )
+        np.testing.assert_equal(len(alpha.couplings_thresh), 3)
+        np.testing.assert_equal(len(alpha.thresh), 3)
+
+
+def test_couplings_exa():
+    "benchmark the exact running of alpha with EKO"
+    test_theory = API.theoryid(theoryid=THEORY_QED)
+    theory = test_theory.get_description()
+    mass_list = [theory["mc"], theory["mb"], theory["mt"]]
+
+    alpha = Alpha(theory, 1e8)
+    couplings = CouplingsInfo.from_dict(
+        dict(
+            alphas=theory["alphas"],
+            alphaem=theory["alphaqed"],
+            scale=theory["Qref"],
+            num_flavs_ref=None,
+            max_num_flavs=theory["MaxNfAs"],
+            em_running=True,
+        )
     )
-    np.testing.assert_almost_equal(
-        alpha.alpha_thresh[3],
-        alpha.alpha_em_fixed_flavor(theory["mc"], alpha.alpha_thresh[4], theory["mb"], 4),
+    eko_alpha = Couplings(
+        couplings,
+        (theory["PTO"] + 1, theory["QED"]),
+        method=CouplingEvolutionMethod.EXACT,
+        masses=[m**2 for m in mass_list],
+        hqm_scheme=QuarkMassScheme.POLE,
+        thresholds_ratios=[1.0, 1.0, 1.0],
     )
-    np.testing.assert_equal(len(alpha.alpha_thresh), 3)
-    np.testing.assert_equal(len(alpha.thresh), 3)
+    coupl_ref = [theory["alphas"], theory["alphaqed"]]
+    for q in [80, 10, 5]:
+        np.testing.assert_allclose(
+            alpha.couplings_fixed_flavor(q, coupl_ref, theory["Qref"], 5)[1],
+            alpha.couplings_variable_flavor(q)[1],
+            rtol=1e-10,
+        )
+        np.testing.assert_allclose(
+            alpha.couplings_fixed_flavor(q, coupl_ref, theory["Qref"], 5),
+            eko_alpha.compute_exact_alphaem_running(
+                np.array(coupl_ref) / (4 * np.pi), 5, theory["Qref"] ** 2, q**2
+            )
+            * 4
+            * np.pi,
+            rtol=1e-7,
+        )
+    for q in [4, 3, 2, 1]:
+        np.testing.assert_allclose(alpha.couplings_variable_flavor(q)[1], eko_alpha.a_em(q**2) * 4 * np.pi, rtol=1e-7)
+    for nf in range(3, theory["MaxNfAs"]):
+        np.testing.assert_allclose(
+            alpha.couplings_thresh[nf],
+            eko_alpha.a(mass_list[nf - 3] ** 2, nf) * 4 * np.pi,
+            rtol=2e-7,
+        )
+
+def test_exa_interpolation():
+    ""
+    test_theory = API.theoryid(theoryid=THEORY_QED)
+    theory = test_theory.get_description()
+
+    alpha = Alpha(theory, 1e8)
+    for q in np.geomspace(1., 1e4, 100, endpoint=True):
+        np.testing.assert_allclose(alpha.couplings_variable_flavor(q)[1], alpha.alpha_em(q), rtol=1e-7)
+
+
+def test_couplings_trn():
+    "benchmark the exact running of alpha with EKO"
+    test_theory = API.theoryid(theoryid=THEORY_QED)
+    theory = test_theory.get_description()
+    theory["ModEv"] = 'TRN'
+    alpha = Alpha(theory, 1e8)
+    coupl_ref = [theory["alphas"], theory["alphaqed"]]
+
+    for q in [80, 10, 5]:
```

Comment:
I would skip this since my fear is that some values of q would make the test fail. For example in `test_exa_interpolation` I had to manually increase the error until I found the one for which the tests were passing, but  maybe there are some values for which the error is larger.
Anyway, with the last commit this PR is done